akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/Split.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.stream.operators.sourceorflow;
+
+import akka.actor.ActorSystem;
+import akka.japi.Pair;
+import akka.japi.function.Creator;
+import akka.japi.function.Function;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Collections;
+
+public class Split {
+  public static void splitWhenExample(String[] args) {
+    ActorSystem system = ActorSystem.create();
+
+    // #splitWhen
+    Source.range(1, 100)
+        .throttle(1, Duration.ofMillis(100))
+        .map(elem -> new Pair<>(elem, Instant.now()))
+        .statefulMapConcat(
+            () -> {
+              return new Function<Pair<Integer, Instant>, Iterable<Pair<Integer, Boolean>>>() {
```

Comment:
`new` needed because lambda can't change variables

Reply:
I've seen that a common Java-land idiom for that is a one element array container. But I think this is just as good (or better).